app/uk/gov/hmrc/cgtpropertydisposals/service/SubscriptionService.scala

```diff
@@ -53,4 +69,96 @@ class SubscriptionServiceImpl @Inject()(connector: SubscriptionConnector)(implic
       }
     }
 
+  override def getSubscription(cgtReference: CgtReference)(
+    implicit hc: HeaderCarrier
+  ): EitherT[Future, Error, SubscriptionDisplayResponse] =
+    connector.getSubscription(cgtReference).subflatMap { response =>
+      lazy val identifiers =
+        List(
+          "id"                -> cgtReference.value,
+          "DES CorrelationId" -> response.header("CorrelationId").getOrElse("-")
+        )
+
+      if (response.status === 200)
+        response
+          .parseJSON[DesSubscriptionDisplayDetails]()
+          .flatMap(toSubscriptionDisplayRecord(_, cgtReference))
+          .leftMap(Error(_, identifiers: _*))
+      else {
+        Left(Error(s"call to subscription display api came back with status ${response.status}"))
+      }
+    }
+
+  def toSubscriptionDisplayRecord(
+    desSubscriptionDisplayDetails: DesSubscriptionDisplayDetails,
+    cgtReference: CgtReference
+  ): Either[String, SubscriptionDisplayResponse] = {
+
+    val desNonIsoCountryCodes: List[CountryCode] =
+      config.underlying.get[List[CountryCode]]("des.non-iso-country-codes").value
+
+    val a = desSubscriptionDisplayDetails.subscriptionDetails.addressDetails
+
+    val addressValidation: Validation[Address] = {
```

Comment:
the same code is being used in the `BusinessPartnerRecordService`. What would be nice is if we factored this out somehow, maybe something like:
```scala
object Address{
  def fromDesAddressDetails(a: DesAddressDetails)(desNonIsoCountryCodes: List[CountryCode]): ValidatedNel[String,Address] = ???
}
```